react-app/components/ListImages/components/ImageItem/index.jsx

```diff
@@ -0,0 +1,57 @@
+import React, { useContext, useState, memo } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faTrash } from '@fortawesome/free-solid-svg-icons';
+import PropTypes from 'prop-types';
+import { getItemLocalStorage, setItemLocalStorage } from '@utils/localStorage';
+import { GalleryContext } from '@contexts/galleryContext';
+
+import './styles.css';
+
+function ImageItem(props) {
+  const { id: idImageItem, url, name, size } = props;
+  const [isHovered, setIsHovered] = useState(false);
+  const { setIsGalleryOpened, setFilesToUploadList } = useContext(GalleryContext);
+
+  const handleMouseEnter = () => {
+    setIsHovered(true);
+  };
+
+  const handleMouseLeave = () => {
+    setIsHovered(false);
+  };
+
+  const handleOpenGallery = () => {
+    setIsGalleryOpened(true);
+  };
+
+  const handleDeleteFile = () => {
+    const filesToUploadListStorage = getItemLocalStorage('filesToUploadList') || [];
+    const filesToUploadListUpdated = filesToUploadListStorage.filter(({ id }) => id !== idImageItem);
+    setItemLocalStorage('filesToUploadList', filesToUploadListUpdated);
+    setFilesToUploadList(filesToUploadListUpdated);
+  };
+
+  return (
+    <div className="card" onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave}>
+      <div className={`title-top ${isHovered ? 'visible' : ''}`} onClick={handleOpenGallery}>
+        {name}
+      </div>
+      <img className={`card-image ${isHovered ? 'faded' : ''}`} src={url} alt={`${name}`} />
+      <div className={`bottom-content ${isHovered ? 'visible' : ''}`}>
+        <div className="title-bottom">{size}</div>
+        <div className="delete-icon" onClick={handleDeleteFile}>
+          <FontAwesomeIcon icon={faTrash} />
+        </div>
+      </div>
+    </div>
+  );
+}
+
+ImageItem.propTypes = {
+  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
```

Comment:
what are we gonna use as the ID? This is what the backend returns for an image:
![image](https://github.com/eduNEXT/xblock-imagesgallery/assets/64440265/bd5d0303-5206-43fb-8d0a-12e7cedd252d)

Is that ID okay?

Reply:
Any string will be ok, make sure that they are not repeated